src/main/java/com/synopsys/integration/detect/workflow/componentlocationanalysis/ScanResultToComponentListTransformer.java

```diff
@@ -29,44 +28,46 @@ public class ScanResultToComponentListTransformer {
      */
     public Set<Component> transformScanResultToComponentList(List<DeveloperScansScanView> rapidScanFullResults) {
         HashMap<String, ScanMetadata> componentIdWithMetadata = new HashMap<>();
-        Set<String> componentNamesWithNullIds = new HashSet<>(rapidScanFullResults.size());
+        Set<String> orderedComponentIDs = new LinkedHashSet<>();
+        Set<String> componentNamesWithNullIds = new LinkedHashSet<>(rapidScanFullResults.size());
```

Comment:
Use of LinkedHashSet for componentNamesWithNullIds so that we display the warnings in the same order as the input for better UX.